src/main/webapp/app/text-editor/text-editor.component.ts

```diff
@@ -111,6 +113,22 @@ export class TextEditorComponent implements OnInit {
         );
     }
 
+    ngOnDestroy() {
+        if (this.canDeactivate() && this.textExercise.id) {
+            if (!this.submission) {
+                this.submission = new TextSubmission();
+            }
+            this.submission.submitted = false;
```

Comment:
There is no point in setting any of the component properties in ngOnDestroy as they will be deleted anyway. Just to make the intention clearer, I would create a throwaway `let` and don't assign anything in the subscribe.

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/971/commits/67e0295ce1158503d63d21d9bf6a3221531fbe6a